workspace/exercise_test.go

```diff
@@ -42,3 +42,19 @@ func TestNewFromDir(t *testing.T) {
 	assert.Equal(t, "the-track", exercise.Track)
 	assert.Equal(t, "the-exercise", exercise.Slug)
 }
+
+func TestMigrateLegacyMetadataFile(t *testing.T) {
```

Comment:
Since this is exported and dangerous (os.Rename & os.Remove) I imagine we want to unit test this, right?

These tests would need to setup both fake metadata (legacy/modern). `writeFakeSolution()` currently lives here https://github.com/exercism/cli/blob/32d8ff8c016e78ab49068681e084901d31a58056/cmd/submit_test.go#L426 and gets used a few places. I tried putting it in `exercise_test.go` and exporting it but that gives errors back in the `cmd` package. I have little experience with Go testing. If we want to share `WriteFakeSolution()` should we put it in `workspace/workspace.go` or would you suggest something else?